docs/reproducibility.md

```diff
@@ -8,48 +8,152 @@ The assumption here is that the original build was done using a Docker.
 
     ```sh
     $ curl -s --cacert cacert.pem https://<LEDGER-URL>/app/version | jq ".version"
-    "1.2.3"
+    "0.12.3-0-gaaaaaaa"
     ```
 
-### Extract measurements
+- Security policy used to verify the container image ([ccf docs](https://microsoft.github.io/CCF/main/governance/gov_api_schemas/2024-07-01.html#get--gov-service-join-policy)), it will contain image layers, e.g.:
+
+    ```sh
+    $ curl -s --cacert cacert.pem https://<LEDGER-URL>/gov/service/join-policy?api-version=2024-07-01 > service-join-policy.json
+    ```
+
+### Extract measurements from the report
 
 The quote contains the attestation report that has the necessary measurements. `Measurement` will have the launch measurement of the guest virtual machine (aka utility VM or UVM) which was used to run the container. `Host data` will have the security policy measurement which was used to make sure no other container was launched except the one you are inspecting. More detail on implementing relying party logic can be [found in Confidential ACI scheme documentation](https://github.com/microsoft/confidential-aci-examples/blob/main/docs/Confidential_ACI_SCHEME.md#reference-info-base64).
 
 - Decode the report and save it to a file:
 
     ```sh
-    cat node-quote.json | jq -r '.raw' | base64 -d > snp-report.bin
+    $ cat node-quotes.json | jq -r '.quotes[0].raw' | base64 -d > snp-report.bin
     ```
 
-- Inspect the report to get the measurements. You could use https://github.com/virtee/snpguest to display the report details:
+- Inspect the report to get the measurements. You could use https://github.com/virtee/snpguest to display the report details, e.g.:
 
     ```sh
-    snpguest display report snp-report.bin
+    $ curl -LO https://github.com/virtee/snpguest/archive/refs/tags/v0.8.0.tar.gz
+    $ tar -xvf v0.8.0.tar.gz
+    $ cd snpguest-0.8.0
+    $ cargo build -r
+    $ cd ..
+    $ ./snpguest-0.8.0/target/release/snpguest display report snp-report.bin
+    <...>
+    Measurement:
+    18 25 a4 bf 2a 9c 38 35 66 a7 17 63 26 83 9a c0
+    e3 6a 1c 5b 37 e9 e6 fa bc 8f dd 71 30 d5 8c ef
+    56 f4 34 75 02 b9 47 89 53 0c ec 19 8a a5 15 43
+
+    Host Data:
+    5a e7 b1 4e e0 c9 c4 fe 26 7d 19 1f 25 b2 0f ff
+    e2 4e 29 c4 ac 41 9c 50 50 1d 20 c8 69 bb ba 65
+    <...>
     ```
 
+- You could also verify the provided report with services such as Microsoft Azure Attestation Service, this step is excluded for the brevity reasons
+
+### Extract image layers from security policy
+
+Inspect the service join policy content and extract the Rego policy used to validate the container:
+
+```sh
+$ cat service-join-policy.json | jq -r '.snp.hostData["5ae7b14ee0c9c4fe267d191f25b20fffe24e29c4ac419c50501d20c869bbba65"]' | printf "%s" "$(cat)" > ccepolicy.rego
```

Comment:
There could be multiple accepted security policies in the join policy (especially during code upgrades). How do customers know which one they should verify? I guess they might need to repeat the same process for every policy in the list until they find one matching the host data present in the attestation report.

Reply:
If the intention is to audit the running service, they all need to be checked. If the intention is to verify a single node, in principle they could start from `/quote/self` and check that, but the load balancer will make that somewhat meaningless.

We could add a `/quote/{node_id}` to facilitate auditing a specific node, but I think in practice it is simpler and more thorough to recommend auditing all the nodes.

Reply:
The intention here is to reproduce the same layers that are in the policy. The policy seems to be keyed by its digest which in turn is in the report under Host data. Correct me if I am wrong but you need only one policy to inspect to get the layers.

Reply:
added a note to use the host data measurement to get the policy